analysis.py

```diff
@@ -191,61 +160,88 @@ def write_statistics(output_directory, summary):
     with open(f"{output_directory}/{DEFAULT_STATISTICS_FILENAME}", "w") as text_file:
         text_file.write(output_content)
 
+    with open(f"{output_directory}/{DEFAULT_RAW_STATS_FILENAME}", "w") as f:
+        json.dump(raw_stats, f, indent=2)
+
+
+def open_node_logs(node_log_glob: str) -> List[str]:
+    node_logs = []
+    for fn in glob(node_log_glob):
+        if fn.endswith("gz"):
+            with gzip.open(fn, "r") as file:
+                node_logs.append(file.read().decode())
+        else:
+            with open(fn, "r") as file:
+                node_logs.append(file.read())
+    return node_logs
+
 
-def fill_rows(task_brackets, stripped_content):
-    filled_rows = dict()
-    gantt_rows = []
-    csv_rows = []
-    table_rows = []
+def count_log_occurences(key: str, node_logs: List[str]) -> int:
+    log_occurences = 0
+    for logfile in node_logs:
+        if key in logfile:
+            log_occurences += 1
+    return log_occurences
 
-    for task_bracket in task_brackets:
-        task_start_item = stripped_content[task_bracket[0]]
-        task_finish_item = stripped_content[task_bracket[1]]
-        task_body = task_start_item.json["task"].split(":", 1)
-        task_id = task_start_item.json["id"]
+
+def fill_rows(content: List[Content], node_logs: List[str]) -> Dict[str, List[Any]]:
+    filled_rows: Dict[str, Any] = dict()
+    gantt_rows: List[Dict[str, Any]] = []
+    csv_rows: List[CSVRow] = []
+    table_rows: List[Dict[str, Any]] = []
+
+    for num, task in enumerate(content):
+        task_body = task.json["task"].split(":", 1)
         task_type = task_body[0].replace("<", "").strip()
         task_desc = task_body[1].replace(">", "").strip()
-        task_body_json = json.loads(task_desc.replace("'", '"'))
-        duration = calculate_duration(task_start_item.timestamp, task_finish_item.timestamp)
+        task_body_json = yaml.safe_load(task_desc.replace("'", '"'))
+
+        # Skip WaitTask
+        if not isinstance(task_body_json, dict):
+            continue
+        duration = task.json["runtime"]
+        # if "id" in task.json:
```

Comment:
Can this be removed?

Reply:
This is left in here because it needs some discussion. https://github.com/raiden-network/scenario-player/pull/535 added logging of the `Task.id` value back into the scenario player's log. Using the `enumerate` id keeps partial compatibility with older log formats. Using the actual `Task.id` value on the other hand will allow for better debugging when looking at e.g. a particularly bad performance of a certain task, because the reported id value will match the id value used in the scenario player log.
What is your opinion?

Reply:
Note: I un-commented this, because I convinced myself for now, that this is the better behavior. I am still interested in your take on this!

Reply:
I think we do not need to care about backwards-compatibility right now. If it makes debugging easier than it's worth it.